Applications/LoRATrainingExample/ContentView.swift

```diff
@@ -199,45 +209,47 @@ class LoRAEvaluator {
             return
         }
 
-        let optimizer = Adam(learningRate: learningRate)
-        try await LoRATrain.train(
-            model: model, train: train, validate: valid, optimizer: optimizer, tokenizer: tokenizer,
-            parameters: parameters
-        ) { progress in
-            await MainActor.run {
-                switch progress {
-                case .train(let i, _, _, _):
-                    self.progress = .init(
-                        title: "Train", current: Double(i), limit: Double(parameters.iterations))
-                case .validation:
-                    output += "\n"
-                default:
-                    break
+        try await modelContainer.perform { model, tokenizer in
```

Comment:
These changes are similar to the ones in LoraCommands (command line version of this)